commit 9b919ad3f99f689c169c7fbd9ba7ac164f51c23b
Author: Huy Nguyen <huyn@mellanox.com>
Date:   Thu Jul 13 13:45:11 2017 -0500

    net/mlx5e: Check for qos capability in dcbnl_initialize
    
    
    [ Upstream commit 33c52b6718d2a6cb414440c98560818910d896dc ]
    
    qos capability is the master capability bit that determines
    if the DCBX is supported for the PCI function. If this bit is off,
    driver cannot run any dcbx code.
    
    Fixes: e207b7e99176 ("net/mlx5e: ConnectX-4 firmware support for DCBX")
    Signed-off-by: Huy Nguyen <huyn@mellanox.com>
    Reviewed-by: Parav Pandit <parav@mellanox.com>
    Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/en_dcbnl.c b/drivers/net/ethernet/mellanox/mlx5/core/en_dcbnl.c
index 2eb54d36e16e..810b51029c7f 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/en_dcbnl.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en_dcbnl.c
@@ -754,6 +754,9 @@ void mlx5e_dcbnl_initialize(struct mlx5e_priv *priv)
 {
 	struct mlx5e_dcbx *dcbx = &priv->dcbx;
 
+	if (!MLX5_CAP_GEN(priv->mdev, qos))
+		return;
+
 	if (MLX5_CAP_GEN(priv->mdev, dcbx))
 		mlx5e_dcbnl_query_dcbx_mode(priv, &dcbx->mode);
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1022_9b919ad3f99f689c169c7fbd9ba7ac164f51c23b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1022_9b919ad3f99f689c169c7fbd9ba7ac164f51c23b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
